### PR TITLE
improve rollup config

### PR DESCRIPTION
**Pull request by kotarella1110**

### CODE REVIEW

**Comment by kotarella1110** [comment]
@bluebill1049 done 👍

File: rollup/writeCjsEntryFile.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const ts = require('typescript');
+const fs = require('fs-extra');
+const path = require('path');
+
+const name = 'index';
```
Comment (bluebill1049) [inline rollup/writeCjsEntryFile.js:6]: not sure if we should keep this in package.json as well, consider we have source in package.json.

**Review by bluebill1049 — COMMENTED** [review]
Awesome stuff! shall we release this as a patch. Less risk for v6 major release.

File: package.json [hunk 2]
```diff
@@ -3,11 +3,12 @@
...
   "main": "dist/index.js",
-  "main:ie": "dist/index.ie11.js",
-  "umd:main": "dist/index.umd.js",
-  "unpkg": "dist/index.umd.min.js",
   "module": "dist/index.esm.js",
-  "module:min": "dist/index.min.esm.js",
+  "umd:main": "dist/index.umd.production.min.js",
+  "unpkg": "dist/index.umd.production.min.js",
+  "jsdelivr": "dist/index.umd.production.min.js",
+  "jsnext:main": "dist/index.esm.js",
```
[the remark this answers is not in the document]
Comment (bluebill1049) [reply package.json:10]: I mean `jsdelivr` `jsnext:main` to more standard naming?

File: rollup/createRollupConfig.js [hunk 3]
```diff
@@ -59,7 +59,13 @@ export function createRollupConfig(opts, assignedPlugins = []) {
           'process.env.NODE_ENV': JSON.stringify(opts.env),
         }),
       sourcemaps(),
-      shouldMinify && terser(),
+      shouldMinify &&
+        terser({
+          output: { comments: false },
+          compress: {
+            drop_console: true,
```
Comment (bluebill1049) [inline rollup/createRollupConfig.js:66]: awesome!

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (bluebill1049) [reply package.json:10]: never mind, i didn't know about this: https://github.com/jsforum/jsforum/issues/5 ignore my comments.

**Comment by bluebill1049** [comment]
looks like there is a warning @kotarella1110

File: rollup/createRollupConfig.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    .join('.');
+
+  return {
+    input: opts.input,
+    output: {
+      file: outputName,
+      format: opts.format,
+      name: umdName,
+      sourcemap: true,
+      globals: { react: 'React', 'react-native': 'ReactNative' },
```
[the remark this answers is not in the document]
Comment (bluebill1049) [reply rollup/createRollupConfig.js]: ignore, you have updated it.

**Comment by bluebill1049** [comment]
> Screen Shot 2020-07-01 at 4 34 58 pm
> 
> looks like there is a warning @kotarella1110

nvm, this warning is only on my local.

File: package.json [hunk 5]
```diff
@@ -93,7 +92,7 @@
   },
   "bundlesize": [
     {
-      "path": "./dist/index.min.esm.js",
+      "path": "./dist/index.cjs.production.min.js",
```
[the remark this answers is not in the document]
Comment (bluebill1049) [reply package.json:95]: ignore as well, not show stopper.

File: rollup/writeCjsEntryFile.js (hunk 1 above)
Comment (bluebill1049) [reply rollup/writeCjsEntryFile.js:6]: sort out later.

**Review by bluebill1049 — APPROVED** [review]
Let's go!

File: package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (kotarella1110) [reply package.json:95]: I think esm build should not minify.
please read here: https://github.com/formik/tsdx/issues/155

File: rollup/createRollupConfig.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      typescript({
+        tsconfig: opts.tsconfig,
+        clean: true,
+      }),
+      resolve(),
+      opts.format === 'umd' &&
+        commonjs({
+          include: /\/node_modules\//,
+        }),
+      opts.env !== undefined &&
```
[the remark this answers is not in the document]
Comment (kotarella1110) [reply rollup/createRollupConfig.js]: Do nothing when it is undefined.

File: package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (bluebill1049) [reply package.json:95]: sounds good 👍 @kotarella1110

**Pull request merged**